internals/daemon/daemon.go

```diff
@@ -702,7 +703,10 @@ func (d *Daemon) doReboot(sigCh chan<- os.Signal, waitTimeout time.Duration) err
 
 const rebootMsg = "reboot scheduled to update the system"
 
-var rebootHandler = systemdModeReboot
+var (
+	rebootHandler         = systemdModeReboot
+	fallbackRebootHandler = systemdModeReboot
```

Comment:
I'm a bit unclear on when `rebootHandler` is used and when `fallbackRebootHandler` is used -- can you clarify?

Reply:
I hope my understanding on this is correct: up until now, Pebble was using `rebootHandler` both for requesting systemd for a system reboot (via `shutdown`) *and* scheduling a *fallback* reboot to happen after `rebootWaitTimeout` (~10min, also via `shutdown`). I think the rationale behind this is, if Pebble fails to immediately reboot, the system wouldn't be stuck running forever, and a reboot is forced after that time window.

When introducing an external reboot mode that essentially does nothing actively to reboot the machine, `rebootHandler` is no longer useful for scheduling this fallback reboot, so what I'm doing is having two reboot handlers now, which for systemd/syscall reboot are one and the same. For external reboot mode though, the `rebootHandler` is set to this "return a special error and do nothing" operation, while the `fallbackRebootHandler` is set to `syscallModeReboot`. If the caller fails to detect `ErrRestartExternal` and reboot the machine, Pebble will do it after `rebootWaitTimeout` has elapsed.